prisma/migrations/20240711071304_add_missing_field_for_api_key/migration.sql

```diff
@@ -0,0 +1,27 @@
+/*
+  Warnings:
+
+  - You are about to drop the column `userId` on the `ApiKey` table. All the data in the column will be lost.
+  - Added the required column `companyId` to the `ApiKey` table without a default value. This is not possible if the table is not empty.
+  - Added the required column `membershipId` to the `ApiKey` table without a default value. This is not possible if the table is not empty.
+
+*/
+-- DropIndex
+DROP INDEX "ApiKey_userId_active_idx";
+
+-- DropIndex
+DROP INDEX "ApiKey_userId_idx";
+
+-- AlterTable
+ALTER TABLE "ApiKey" DROP COLUMN "userId",
+ADD COLUMN     "companyId" TEXT NOT NULL,
+ADD COLUMN     "membershipId" TEXT NOT NULL;
+
+-- CreateIndex
+CREATE INDEX "ApiKey_membershipId_idx" ON "ApiKey"("membershipId");
+
+-- CreateIndex
+CREATE INDEX "ApiKey_membershipId_active_idx" ON "ApiKey"("membershipId", "active");
+
+-- CreateIndex
+CREATE INDEX "ApiKey_companyId_idx" ON "ApiKey"("companyId");
```

Comment:
Here is the reason why I scoped the API to user level

* So `Super admin` can get collection of `/companies`
* They can manage other companies as well with same API, especially useful for partnerships and integrations.


I guess we can control this API access from role `member` table as well. But having it scoped to user level would avoid having to create multiple API keys per company.

Reply:
- It's a security risk having a user-level API key. If one gets leaked, all other companies are compromised.

- If a membership is deleted, the key will automatically become unusable, but with a user, it won't, since users don't get deleted, memberships do.

- Having a user relationship rather than the membership is going to involve some complex queries and checks to determine whether the user is a member.

Reply:
> * It's a security risk having a user-level API key. If one gets leaked, all other companies are compromised.
> * If a membership is deleted, the key will automatically become unusable, but with a user, it won't, since users don't get deleted, memberships do.
> * Having a user relationship rather than the membership is going to involve some complex queries and checks to determine whether the user is a member.

Makes sense, let's go with this flow in that case.